packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -419,11 +419,13 @@ class ChatOpenAIProvider(BaseProvider, OpenAIChat):
     id = "openai-chat"
     name = "OpenAI"
     models = [
-        "gpt-4",
         "gpt-4-0314",
-        "gpt-4-32k",
-        "gpt-4-32k-0314",
         "gpt-3.5-turbo",
+        "gpt-4-0613",
+        "gpt-4",
+        "gpt-3.5-turbo-16k-0613",
+        "gpt-3.5-turbo-16k",
+        "gpt-3.5-turbo-0613",
         "gpt-3.5-turbo-0301",
```

Comment:
Taking a look at the [available GPT models on OpenAI](https://platform.openai.com/docs/models/gpt-4), it appears this branch is missing the following models:

- `gpt-4-32k`
- `gpt-4-32k-0613`
- `gpt-4-0314`
- `gpt-4-32k-0314`

These models should be added to both providers. Also, for readability, could you sort this list in the code?

Reply:
I believe the `gpt-4` models are deprecated to be used on the non-chat OpenAI LLM, can we check if these work with OpenAI provider.

Reply:
@3coins 
```
import openai
models = dict(openai.Model.list())
for i in models['data']:
    if i['id'].startswith('gpt'):
        print(i['id'])
```
gpt-4-0314
gpt-3.5-turbo
gpt-4-0613
gpt-4
gpt-3.5-turbo-16k-0613
gpt-3.5-turbo-16k
gpt-3.5-turbo-0613
gpt-3.5-turbo-0301

Reply:
I meant testing these with the OpenAI LangChain LLM class.

Reply:
Here is the reference from LangChain repo.
https://github.com/langchain-ai/langchain/blob/19f504790ea9e8df4f6613003379d5099c915115/libs/langchain/langchain/llms/openai.py#L197C10-L197C10

Reply:
Like this https://github.com/jupyterlab/jupyter-ai/issues/336#issuecomment-1675144108

Reply:
@3coins Both `OpenAIChat` (inheriting from `BaseLLM`) and `ChatOpenAI` (inheriting from `BaseChatModel`) pass the model ID directly to the underlying `openai.ChatCompletion` client. The warning is really regarding usage of `OpenAIChat`, not regarding usage of GPT-3.5 or GPT-4 models with `OpenAIChat`.